### DISCUSSION TITLE
PressReader files

### DISCUSSION THREAD

**Discussion by Commodore64user** [koreader/koreader#11791] [Q&A] [answered]
Sometimes I like to download some publications from PressReader, however when they do the conversion [PressReader website] from PDF (or whatever file they use) to ePub, if an article spreads over "n" pages, it is rendered "n" times.

So in this example, the article "How to rescue Europe" starts on page 26 of the PDF and continues on page 27, where it occupies about half that page. Then "Chinese whisperers" starts on the second half of page 27 and ends in page 28. "Odessa's vanishing lads" then occupies just the final half of page 28. As you can see in the ToC they are rendered 2, 2 and 1 times respectively

<img src="https://github.com/koreader/koreader/assets/97603719/7fddf5d5-2380-454e-909e-82b865c5b7e3" width=50% height=50%>

My question is, Is there some sort of "style tweak" I could use to match the titles of the articles and hide all the repetitions? Ideally I would like to add it to a profile. Thanks!

edit: also, I would like to hide a specific chapter called "Contents", it always has the same name.

edit 2: adding screenshot of the print-matched PDF version (which is not available for download).

<img src="https://github.com/koreader/koreader/assets/97603719/f91ee024-0547-4394-ae42-e5cdc400adf9" width=50% height=50%>

Edit 3: here is a clearer explanation of what i need: Essentially it is an automated hidden flow that first removes from the flow the ToC-level-1 chapter "Contents" (this bit is important as the conversion also duplicates articles shown on the magazine's ToC -original file- via links, hope that is clear enough..?) and then checks all the ToC-level-3 ones and hides the ones with similar names (i.e repetitions) keeping only one copy of each article. This would ideally be triggered via profiles or could be applied (on opening) to all files that start with the name "Name of newspaper/magazine"

**Comment by Frenzie** [comment]
I think "if it has the same name"  is a bit too advanced. It might be possible with a user patch but really the problem is the nonsensical ToC itself. "Page 28" isn't something that should be in there. Maybe they have some support?

**Reply by Commodore64user** [reply]
I think they do it that way so, anyone can say. Hey on "page 26" (physical newspaper) there is a great article. And so, you can look at the ToC and see what is in said page. I did try contacting them about it, but their engineers don't seem interested in changing anything as "it works as intended".

**Reply by Commodore64user** [reply]
If you have a library card you can use it to download a test file https://www.pressreader.com (they print you email on _every page_ so won't be sharing any of mine). You need to create an account (and link the library card to it) and to do it from a browser not the app.

**Reply by Frenzie** [reply]
I hadn't really noticed but my newspaper is even worse. :rofl: 
<img src=https://github.com/koreader/koreader/assets/202757/b7b5d4f4-afa4-4ecb-8a4b-5df67f0afbe3 width=40%>

Edit: mostly because they implemented browsing in the document itself:
<img src=https://github.com/koreader/koreader/assets/202757/84556651-d7e0-4ba5-b08a-6c57f9752d9d width=40%> <img src=https://github.com/koreader/koreader/assets/202757/fe0a4eba-b114-40ec-b242-748d5ceee264 width=40%>

**Reply by Commodore64user** [reply]
but is your repeating the articles multiple times?, I sometimes have to skip three chapters before landing on the proper next article.

I don't really mind the ToC, all I want is to hide the reps from the flow. kind of like a hidden flows but that occurs automagically.

**Reply by Frenzie** [reply]
Nothing is repeated obviously, but I'm not sure what you mean by repeating regardless. Are you saying they're actually repeating the article in full multiple times rather than just the ToC entry?

**Reply by Commodore64user** [reply]
yes, that is exactly what I am saying. they repeat the articles "n" times in full. so a three page article, is printed three times in full, usually back to back (but not always as is the case with "Chinese whisperers")

**Reply by Commodore64user** [reply]
Okay so i have developed the description of what i want/need a little further.

As you said, this has user patch written all over it.

Essentially it is an automated hidden flow that first removes from the flow the ToC-level-1 chapter "Contents" (this bit is important as the conversion also duplicates articles shown on the magazine's ToC -original file- via links, hope that is clear enough..?) and then checks all the ToC-level-3 ones and hides the ones with similar names (i.e repetitions) keeping only one copy of each article.

This would ideally be triggered via profiles or could be applied (on opening) to all files that start with the name "Name of newspaper/magazine"

How can i make it happen?

**Reply by Frenzie** [reply]
I'm not sure if there's a sane way to do that. Are you sure they didn't just misunderstand the same way I did when they said working as intended? A ToC entry for every "page" an article is on is a bit odd, but it's very different from repeating the article itself multiple times.

**Reply by Commodore64user** [reply]
No, they were quite clear about not wanting to touch how it works. I guess is more related to, it works good enough and they don't want to waste engineer's time (and company money) over it. Also, I would imagine most people use the app so I am probably in the minority that downloads the e-reader version.

It does make 'some' sense when you understand how they approach the problem, with the consequence that articles are repeated multiple times, the alternative would be way worse, that some articles or portions of articles are not printed at all. 

Essentially the level-2 in ToC is there to help you know where an article is found in the physical version. What they do in the conversion process is: first show you a small picture of the original page and then, print every article found on said page in full, rinse and repeat for every page. That guarantees that someone reading the physical paper can say the article on page 26 and the person with the ePub can find it. This is particularly true if in the case of newspapers the article begins in the front page but continues on page 17, if someone says to you, the article is on page 17 but fails to mention where it begins you are not left wondering where it starts, or you don't have to jump around following links. The side effect again is that articles are printed multiple times when they spread over multiple pages.

Doing it that way, you are guaranteed not to miss anything but leaves me trying to solve this now...

If you have a library card you should definitely check it out, there are lots publications also it would make this conversation a lot simpler, ;) and even if you don't have one, go to a hotel (or other place listed on their app/website, they call them hotspots) as most offer free passes when you connect to their wifi

**Reply by Commodore64user** [reply]
One more thing about their ToC: level 1 is the general sections, say "Technology" or "Europe", or "Britain", level 2 is page (from PDF/physical) and level 3 is the name of article.

**Reply by Frenzie** [reply]
> would make this conversation a lot simpler

The problem is clear. It's just a very strange and unusual problem.

**Comment by jonnyl2** [comment]
It's a crapshoot, but have you tried the alternative TOC? It sometimes works wonders.

Edit: Nvm, it's an ePub converted from PDF, so probably little chance of it being effective.

**Comment by Commodore64user** [comment, marked as answer]
Excellent news, this problem has been finally solved. https://github.com/Commodore64user/pressreader_epub_deduplicator
